### PR TITLE
fix(rpc): chain-id in grpc query is not initialized without abci event

### PR DESCRIPTION
**Pull request by mmsqe**
for more info:
https://github.com/evmos/ethermint/pull/1405
https://github.com/evmos/ethermint/pull/1488

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/integration_tests/test_grpc_only.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    api_port = ports.api_port(custom_ethermint.base_port(1))
+    # in normal mode, grpc query works even if we don't pass chain_id explicitly
+    success = False
+    max_retry = 3
+    sleep = 1
+    for i in range(max_retry):
+        rsp = grpc_eth_call(api_port, msg)
+        ret = rsp["ret"]
+        valid = ret is not None
+        if valid and 9000 == int.from_bytes(base64.b64decode(ret.encode()), "big"):
```
Comment (mmsqe) [inline tests/integration_tests/test_grpc_only.py:72]: [test](https://github.com/crypto-org-chain/ethermint/actions/runs/4362229790/jobs/7626882060) was not stable, is it ok to merge the test [fix](https://github.com/crypto-org-chain/ethermint/pull/216) together? @yihuang

Comment (yihuang) [reply tests/integration_tests/test_grpc_only.py:72]: sure 👍

**Review by yihuang — APPROVED** [review]

**Comment by yihuang** [comment]
there's a python lint issue.

**Comment by mmsqe** [comment]
> there's a python lint issue.

yea, fixed

**Pull request merged**